src/rust/engine/src/nodes.rs

```diff
@@ -397,18 +487,21 @@ pub struct SelectDependencies {
   pub subject: Key,
   pub variants: Variants,
   pub selector: selectors::SelectDependencies,
+  pub edges: rule_graph::RuleEdges
```

Comment:
Storing the edges in `Selectors` isn't cheap, since it means they end up stored in the `NodeKey` and thus in the `graph::Graph`.

If it's possible to compute/request them from the `context.rule_graph` when they're needed, that would probably be best for now. Failing that, one thing I'd been looking at is keeping `Arc`s of larger values from the `Tasks` struct... might be helpful here as well if these are large.